packages/plugin-dialog/src/ui/DialogZoneDomRenderer.ts

```diff
@@ -0,0 +1,43 @@
+import { DomRenderingEngine } from '../../../plugin-dom/src/DomRenderingEngine';
+import { AbstractRenderer } from '../../../plugin-renderer/src/AbstractRenderer';
+import { DialogZoneNode } from './DialogZoneNode';
+import { VNode } from '../../../core/src/VNodes/VNode';
+import template from './Dialog.xml';
+import './Dialog.css';
+
+const container = document.createElement('jw-container');
+container.innerHTML = template;
+const floatItem = container.firstElementChild;
+
+export class DialogZoneDomRenderer extends AbstractRenderer<Node[]> {
+    static id = 'dom';
+    engine: DomRenderingEngine;
+    predicate = DialogZoneNode;
+
+    async render(node: DialogZoneNode): Promise<Node[]> {
+        const float = document.createElement('jw-dialog-container');
+        for (const child of node.children) {
+            if (!node.hidden.get(child)) {
+                float.appendChild(await this._renderDialog(child));
+            }
+        }
+        return float.childNodes.length ? [float] : [document.createDocumentFragment()];
+    }
+
+    private async _renderDialog(node: VNode): Promise<Element> {
+        const clone = floatItem.cloneNode(true) as Element;
```

Comment:
It might be good to remove the confusing "float" from the code. This `floatItem` *is* a `dialog` now.